src/index.ts

```diff
@@ -76,4 +77,61 @@ app.post("/logout", async(c) => {
   return new Response();
 });
 
+app.post("/attendance", async(c) => {
+  /*ネットワークの確認を行う*/
+  const allowPattern = new RegExp(c.env.ALLOW_IP_REGEX);
+  if ( !allowPattern.test(c.req.header("cf-connectiong-ip") ?? "") ) { //"??":c.req.header(cf-connectiong-ip")でnullが返ってきたときに""として扱う演算子
+    return c.text("Forbidden", 403);
+  }
+
+  /*学生の認証を行う*/
+  //sessionがcookieあるか確認
+  const session = c.get("session");
+  if ( !session ){
+    return c.text("Unauthorized", 401);
+  }
```

Comment:
```suggestion
  const honoSession = c.get("session");
  const session = honoSession.get("login") as Session | null;
  if ( !session ){
    return c.text("Unauthorized", 401);
  }
```
この `c.get("session")` で取得できるのは `hono-sessions` の API のオブジェクトですので, ここからさらに私たちが格納しているセッションのオブジェクトを取り出す必要があります. この `Session` 型は `./model/session` モジュールのものです.

Reply:
90行目での処理について
/loginでセットしたセッションの情報を取得するから引数にloginを入れているのですか？

Reply:
はい, そうです

Reply:
ありがとうございます。